cloudinit/helpers.py

```diff
@@ -333,33 +333,39 @@ class Paths(persistence.CloudInitPickleMixin):
     def __init__(self, path_cfgs: dict, ds=None):
         self.cfgs = path_cfgs
         # Populate all the initial paths
-        self.cloud_dir = path_cfgs.get("cloud_dir", "/var/lib/cloud")
-        self.run_dir = path_cfgs.get("run_dir", "/run/cloud-init")
-        self.instance_link = os.path.join(self.cloud_dir, "instance")
-        self.boot_finished = os.path.join(self.instance_link, "boot-finished")
-        self.seed_dir = os.path.join(self.cloud_dir, "seed")
+        self.cloud_dir: str = path_cfgs.get("cloud_dir", "/var/lib/cloud")
+        self.run_dir: str = path_cfgs.get("run_dir", "/run/cloud-init")
+        self.instance_link: str = os.path.join(self.cloud_dir, "instance")
+        self.boot_finished: str = os.path.join(
+            self.instance_link, "boot-finished"
+        )
+        self.seed_dir: str = os.path.join(self.cloud_dir, "seed")
         # This one isn't joined, since it should just be read-only
-        template_dir = path_cfgs.get("templates_dir", "/etc/cloud/templates/")
-        self.template_tpl = os.path.join(template_dir, "%s.tmpl")
+        template_dir: str = path_cfgs.get(
+            "templates_dir", "/etc/cloud/templates/"
+        )
+        self.template_tpl: str = os.path.join(template_dir, "%s.tmpl")
         self.lookups = {
+            "boothooks": "boothooks",
+            "cloud_config": "cloud-config.txt",
+            "data": "data",
             "handlers": "handlers",
+            "instance_data": "instance-data.json",
```

Comment:
Nit: Maybe it's helpful to carry over the comments from `cloudinit.sources.__init__`:

```suggestion
            # File in which public available instance meta-data is written
            # security-sensitive key values are redacted from this
            # world-readable file
            "instance_data": "instance-data.json",
            # security-sensitive key values are present in this
            # root-readable file
```